src/fd.cpp

```diff
@@ -110,19 +159,99 @@ static int wait_thread(void *arg) {
     std::fill(args->results->begin(), args->results->end(), NA_INTEGER);
   }
 
+  return 0;
+
+}
+
+static int wait_thread_single(void *arg) {
+
+  tct_thrd_detach(tct_thrd_current());
+
+  std::unique_ptr<std::shared_ptr<ThreadArgs>> argsptr(static_cast<std::shared_ptr<ThreadArgs>*>(arg));
+  std::shared_ptr<ThreadArgs> args = *argsptr;
+
+  if (wait_on_fds(args))
+    return 1;
+
   callbackRegistryTable.scheduleCallback(later_callback, static_cast<void *>(argsptr.release()), 0, args->loop);
 
   return 0;
 
 }
 
+static int wait_thread_persistent(void *arg) {
+
+  tct_thrd_detach(tct_thrd_current());
+  thread_active->store(true); // atomic allows update prior to acquiring lock
```

Comment:
Why do this here, instead of in `execLater_launch_thread` right before calling `tct_thrd_create`? If you did that, you might not need the cv at all?

Reply:
It's a synchronisation point. But probably confusing as the atomic is updated outside of the lock. The logic will be clearer by eliminating the atomic as you suggest below.